docs(contributing): add "Good Git and Commit Habits" guide

nina2dv opened the pull request:
- Cover recommended practices to be adopted, such as atomic commits, frequent commits, local history cleanup, meaningful messages, CLI over GUI, and consistent workflows
- Include references and visualization resources

% FOOTER %
type-of: docs

closes: https://esg-yorku-team.atlassian.net/browse/CE-27
closes-desc: "Research: optimal commit habits"

nina2dv committed:
docs(contributing): add "Good Git and Commit Habits" guide

- Cover recommended practices to be adopted, such as atomic commits, 
frequent commits, local history cleanup, meaningful messages, 
CLI over GUI, and consistent workflows
- Include references and visualization resources

% FOOTER %
type-of: docs

closes: https://esg-yorku-team.atlassian.net/browse/CE-27
closes-desc: "Research: optimal commit habits"

liaskos commented on src/site/markdown/contributors/git-habits.md:
```diff
@@ -0,0 +1,147 @@
+> Tip: If you want to commit just because you need a clean working copy (to checkout a branch, pull in changes, etc.),
+> consider `git stash` or `git worktree` (checkout multiple branches from the same repo at the same time)
+
+## Meaningful Commit Messages
```
@nina2dv Please link to the document that describes the commit structure.

nina2dv replied:
I have added:
```
  - Please see our [Commit Conventions and Structure Guide](./commits.md) for more detail
```

liaskos commented on src/site/markdown/contributors/git-habits.md:
````diff
@@ -0,0 +1,147 @@
+```bash
+# Example: Combine the last 3 commits interactively
+git rebase -i HEAD~3
+```
````
Maybe add:

Caution: use `rebase` only if you understand the command and you are on a personal branch. Experiment on a test repository.

nina2dv replied:
I have added:

> Caution: use rebase only if you understand the command and you are on a personal branch.

liaskos reviewed (changes requested):
@nina2dv Nina thank you for this - see individual comments. Can you also add this to the documentation structure in `site.xml`? I will create a ticket for you to draft the "Main Contributor Workflow" as per your research and our discussion Monday

nina2dv committed:
docs(contributing): refine "Good Git and Commit Habits" guide

- Add link to `commits.md` guide
- Add caution note for rebase usage

% FOOTER %
type-of: docs

closes: https://esg-yorku-team.atlassian.net/browse/CE-27
closes-desc: "Research: optimal commit habits"

nina2dv committed:
docs(site): add "Git and Commit Habits" page to `site.xml`

- Link new page under Contributor’s Guide section of `site.xml`

% FOOTER %
type-of: docs

relates-to: https://esg-yorku-team.atlassian.net/browse/CE-27
relates-to-desc: "Research: optimal commit habits"

nina2dv commented:
> @nina2dv Nina thank you for this - see individual comments. Can you also add this to the documentation structure in `site.xml`? I will create a ticket for you to draft the "Main Contributor Workflow" as per your research and our discussion Monday

I added the git/commit habits page to `site.xml` here: https://github.com/cmg-york/cnsim-engine/pull/7/commits/ed31573b66034faa66cab04451def924d78c84b5

nina2dv reviewed (commented):
I have made the 2 requested changes in https://github.com/cmg-york/cnsim-engine/pull/7/commits/c35e546c311dc468fb80ce0fa596418777e11d94

liaskos reviewed (changes requested):
Nina, somehow I keep getting the "details closed" error in ed31573

nina2dv commented:
> Nina, somehow I keep getting the "details closed" error in [ed31573](https://github.com/cmg-york/cnsim-engine/commit/ed31573b66034faa66cab04451def924d78c84b5)

@liaskos Try it now. I have moved my commits onto latest `main`.

liaskos commented:
@nina2dv Nina this works fine now, many thanks. I have one more small thing, but will do a ticket for this.
